app/src/pages/_app.tsx

```diff
@@ -1,45 +1,75 @@
-import { AppContext } from '@context/state'
 import { appWithTranslation } from 'next-i18next'
 import type { AppProps } from 'next/app'
-import useLocalStorage from 'src/hooks/useLocalStorage'
+import { useRouter } from 'next/router'
+import { useEffect } from 'react'
 
 import Layout from '@components/Layout'
+import PageError from '@components/PageError'
 
+import useSessionStorage from '@src/hooks/useSessionStorage'
 import '@styles/styles.scss'
+import { getBackRoute, getForwardRoute, hasRoutingIssues } from '@utils/routing'
+import { initialSessionData } from '@utils/sessionData'
 
 function MyApp({ Component, pageProps }: AppProps) {
-  const [session, setSession] = useLocalStorage('session', {
-    clinic: undefined,
-    contact: {
-      firstName: '',
-      lastName: '',
-      phone: '',
-      comments: '',
-    },
-    eligibility: {
-      residential: '',
-      pregnant: false,
-      baby: false,
-      child: false,
-      guardian: false,
-      none: false,
-      loss: false,
-      before: '',
-      insurance: false,
-      snap: false,
-      tanf: false,
-      fdpir: false,
-      none2: false,
-    },
-  })
+  const sessionKey = 'session'
+  const [session, setSession] = useSessionStorage(
+    sessionKey,
+    initialSessionData
+  )
+
+  // Note: All router.push() calls have linting disabled on them.
+  // See https://nextjs.org/docs/api-reference/next/router#potential-solutions
+
+  // Pass in a prop for whether the form wizard is in review mode.
+  const router = useRouter()
+  const reviewMode = router.query.mode === 'review'
+
+  // Handle form wizard page access.
+  useEffect(() => {
+    try {
+      const outcome = hasRoutingIssues(router.pathname, session)
+      if (outcome.error) {
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        router.push({
+          pathname: outcome.cause,
+          query: { error: 'missing-data' },
+        })
+      }
+    } catch (e: unknown) {
+      const error = e as Error
+      console.log(`error caught: ${error.message}`)
+      // Something bad happened with regard to session. Route to '/' with an error message.
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      router.push({ pathname: '/', query: { error: 'missing-data' } })
+    }
```

Comment:
Notes for reviewers: This shouldn't ever happen, but typescript informed me that `session` could be an empty function, rather than an object, so _if_ that weird case happens, the user gets sent back to the index page and shown the error.